File: pset6/credit/credit.py
from cs50 import get_int

cc = get_int("Number: ")

# get number of digits
digits = 1
ccDigits = cc

while ccDigits / 10 > 1:
    digits += 1
    ccDigits /= 10

# check sum of even digits
evenSum = 0
for i in range(1, digits, 2):
    n1 = int(cc / pow(10, i))
    n2 = int(n1 % 10 * 2)
    while (n2 > 9):
        n2 = int(n2 % 10 + n2 / 10)
    evenSum += int(n2)

# check sum of odd digits
oddSum = 0
for j in range(0, digits, 2):
    j1 = int(cc / pow(10, j))
    j2 = int(j1 % 10)
    oddSum += int(j2)

# total
totalSum = evenSum + oddSum

if totalSum % 10 != 0:
    print("INVALID")

else:
    amex = int(cc / pow(10, 13))
    mastercard = int(cc / pow(10, 14))
    visa1 = int(cc / pow(10, 12))
    visa2 = int(cc / pow(10, 15))

    if digits == 15 and (amex == 34 or amex == 37):
        print("AMEX")

    elif digits == 16 and (51 <= mastercard <= 55):
        print("MASTERCARD")

    elif (digits == 13 and visa1 == 4) or (digits == 16 and visa2 == 4):
        print("VISA")

    else:
        print("INVALID")

File: pset1/credit/credit.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    long cc = get_long("Number: ");
    long cc_digits = cc;

    //get number of digits
    int digits = 1;

    while (cc_digits / 10 > 1)
    {
        digits++;
        cc_digits /= 10;
    } // digits = number of digits

    // check the sum of even digits
    int even_sum = 0;
    for (int i = 1; i < digits; i += 2)
    {
        long n1 = cc / (long) pow(10, i);
        int n2 = n1 % 10;
        n2 *= 2;
        while (n2 > 9)
        {
            n2 = n2 % 10 + n2 / 10;
        }
        int n3 = n2 % 10;
        even_sum += n3;
    }

    // check the sum of odd digits
    int odd_sum = 0;
    for (int i = 0; i < digits; i += 2)
    {
        long n1 = cc / (long) pow(10, i);
        int n2 = n1 % 10;
        odd_sum += n2;
    }

    //check the total sum
    int total_sum = even_sum + odd_sum;

    if (total_sum % 10 != 0)
    {
        printf("INVALID\n");
    }
    else
    {
        long amex = cc / (long) pow(10, 13);
        long mastercard = cc / (long) pow(10, 14);
        long visa1 = cc / (long) pow(10, 12);
        long visa2 = cc / (long) pow(10, 15);

        if (digits == 15 && (amex == 34 || amex == 37))
        {
            printf("AMEX\n");
            return 0;
        }

        if (digits == 16 && mastercard >= 51 && mastercard <= 55)
        {
            printf("MASTERCARD\n");
            return 0;
        }

        if ((digits == 13 && visa1 == 4) || (digits == 16 && visa2 == 4))
        {
            printf("VISA\n");
            return 0;
        }

        else
        {
            printf("INVALID\n");
        }

    }

}
